From 3077ef0086b24f0033b4f7d9e57c68254729ba97 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Wed, 6 Nov 2013 16:07:25 +0800
Subject: [PATCH 11/43] ENGR00215174 update to pass build for android 4.4.

change drawingState() to getDrawingState().
add getContentTransform() function to be used by LayerDim.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 services/surfaceflinger/Layer.cpp          | 4 ++++
 services/surfaceflinger/Layer.h            | 1 +
 services/surfaceflinger/LayerDim.cpp       | 4 ++--
 services/surfaceflinger/SurfaceFlinger.cpp | 1 +
 4 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/services/surfaceflinger/Layer.cpp b/services/surfaceflinger/Layer.cpp
index 5ff79a9..506d5a8 100644
--- a/services/surfaceflinger/Layer.cpp
+++ b/services/surfaceflinger/Layer.cpp
@@ -320,6 +320,10 @@ Rect Layer::getContentCrop() const {
     return crop;
 }
 
+uint32_t Layer::getContentTransform() const {
+    return mCurrentTransform;
+}
+
 static Rect reduce(const Rect& win, const Region& exclude) {
     if (CC_LIKELY(exclude.isEmpty())) {
         return win;
diff --git a/services/surfaceflinger/Layer.h b/services/surfaceflinger/Layer.h
index c1e5e9f..07b5ab8 100644
--- a/services/surfaceflinger/Layer.h
+++ b/services/surfaceflinger/Layer.h
@@ -290,6 +290,7 @@ public:
      * to the layer's size.
      */
     Rect getContentCrop() const;
+    uint32_t getContentTransform() const;
 
     /*
      * Returns if a frame is queued.
diff --git a/services/surfaceflinger/LayerDim.cpp b/services/surfaceflinger/LayerDim.cpp
index 2c9d44e..fd1aa7e 100644
--- a/services/surfaceflinger/LayerDim.cpp
+++ b/services/surfaceflinger/LayerDim.cpp
@@ -74,7 +74,7 @@ Rect LayerDim::computeCrop(const sp<const DisplayDevice>& hw) const {
 
     // the active.crop is the area of the window that gets cropped, but not
     // scaled in any ways.
-    const State& s(drawingState());
+    const State& s(getDrawingState());
 
     // apply the projection's clipping to the window crop in
     // layerstack space, and convert-back to layer space.
@@ -138,7 +138,7 @@ void LayerDim::setGeometry(
     layer.setBlending(HWC_BLENDING_DIM);
 
     // set dim value
-    const State& s(drawingState());
+    const State& s(getDrawingState());
     layer.setPlaneAlpha(s.alpha);
 
     if (isSecure() && !hw->isSecure()) {
diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index bb636c5..9231441 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -1312,6 +1312,7 @@ void SurfaceFlinger::postFramebuffer()
     uint32_t flipCount = getDefaultDisplayDevice()->getPageFlipCount();
     if (flipCount % LOG_FRAME_STATS_PERIOD == 0) {
         logFrameStats();
+    }
 
     if (mDebugFps) {
         debugShowFPS();
-- 
1.8.0

